plugins/woocommerce/src/Blocks/BlockTypes/ProductFilterStatus.php

```diff
@@ -18,7 +18,195 @@ final class ProductFilterStatus extends AbstractBlock {
+					],
+				],
+				'spacing'              => [
+					'margin'                        => true,
+					'padding'                       => true,
+					'blockGap'                      => true,
+					'__experimentalDefaultControls' => [
+						'margin'   => false,
+						'padding'  => false,
+						'blockGap' => false,
+					],
+				],
+				'__experimentalBorder' => [
+					'color'                         => true,
+					'radius'                        => true,
+					'style'                         => true,
+					'width'                         => true,
+					'__experimentalDefaultControls' => [
+						'color'  => false,
+						'radius' => false,
+						'style'  => false,
+						'width'  => false,
+					],
+				],
+			],
+			'attributes'   => [
+				'showCounts'   => [
+					'type'    => 'boolean',
+					'default' => false,
+				],
+				'displayStyle' => [
+					'type'    => 'string',
+					'default' => 'woocommerce/product-filter-checkbox-list',
+				],
+				'isPreview'    => [
+					'type'    => 'boolean',
+					'default' => false,
+				],
+				'hideEmpty'    => [
+					'type'    => 'boolean',
+					'default' => true,
+				],
+				'clearButton'  => [
+					'type'    => 'boolean',
+					'default' => false,
+				],
+			],
+		];
+
+		$block_settings['render_callback'] = $this->get_block_type_render_callback();
+		$block_settings['editor_script']   = $this->get_block_type_editor_script( 'handle' );
+		$block_settings['editor_style']    = $this->get_block_type_editor_style();
+		$block_settings['style']           = $this->get_block_type_style();
+
+		if ( isset( $this->api_version ) && '2' === $this->api_version ) {
+			$block_settings['api_version'] = 2;
+		}
+
+		/**
+		 * We always want to load block styles separately, for every theme.
+		 * When the core assets are loaded separately, other blocks' styles get
+		 * enqueued separately too. Thus we only need to handle the remaining
+		 * case.
+		 */
+		if (
+			! is_admin() &&
+			! wc_current_theme_is_fse_theme() &&
+			$block_settings['style'] &&
+			(
+				! function_exists( 'wp_should_load_separate_core_block_assets' ) ||
+				! wp_should_load_separate_core_block_assets()
+			)
+		) {
+			$style_handles           = $block_settings['style'];
+			$block_settings['style'] = null;
+			add_filter(
+				'render_block',
+				function ( $html, $block ) use ( $style_handles ) {
+					if ( $block['blockName'] === $this->get_block_type() ) {
+						array_map( 'wp_enqueue_style', $style_handles );
+					}
+					return $html;
+				},
+				10,
+				2
+			);
+		}
+
+		if ( $this->stock_status_options ) {
+			foreach ( wc_get_product_stock_status_options() as $key => $value ) {
+				$block_settings['attributes'][ $key ] = [
+					'type'    => 'boolean',
+					'default' => true,
+				];
+			}
+		}
+
+		if ( $this->status_options ) {
+			foreach ( $this->status_options as $key => $value ) {
+				$block_settings['attributes'][ $key ] = [
+					'type'    => 'boolean',
+					'default' => true,
+				];
+			}
+		}
```

Comment:
I don't think we should use this direction. Because 1) Stock statuses are filterable and 2) in the future we might add more status to the filter, they can consider dynamic data, and we shouldn't structure attributes based on dynamic data. Why don't we use a single attribute to hold the enabled statuses? Or two keys, one for stock and one for others?

Reply:
> Do you mean have two attributes for example stockStatuses and productStatuses and then dynamically populate their values?

Yes exactly.

Reply:
Done